airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDatabase.java

```diff
@@ -92,6 +96,15 @@ public static HikariDataSource createDataSource(final JsonNode config) {
       dataSource.setUsername(username);
       dataSource.setPassword(credentials.get("password").asText());
 
+    } else if (credentials != null && credentials.has(PRIVATE_KEY_FIELD_NAME)) {
```

Comment:
I think the following scenario is possible and is a bug:
1. user configures a Snowflake Source to use with username/password authentication
2. user re-configures the same Snowflake Source to use private key
3. Snowflake source will continue to use username and password instead of private key unless the user explicitly clears `password` field

Reply:
No, fortunately it is not a case, because from UI we obtain config just with on option for credentials : password, oauth2 or key pair, and for example: 

1. We use password auth during first sync.
2. For second sync we chose key pair auth and sync with it. During second sync on server side we got following cred object
![Screenshot from 2022-07-14 19-26-01](https://user-images.githubusercontent.com/33333155/179032563-571a3652-0c7f-495e-a487-611bc41ce618.png)

So here we don't have password from first sync in config, password is stored, but not passed to connector if another auth option is chosen.